### PR TITLE
Add .plugins-ml-connector to system index

### PR DESCRIPTION
**Pull request by zane-neo**
### Description
[Describe what this change achieves]
Add .plugins-ml-connector to system index

### Issues Resolved
NA

Is this a backport? If so, please add backport PR # and/or commits #
No

### Testing
NA

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by cwperks — COMMENTED** [review]
@zane-neo Can you also add this to the batch script version of [`install_demo_configuration.bat`](https://github.com/opensearch-project/security/blob/main/tools/install_demo_configuration.bat) as well?

**Comment by cwperks** [comment]
@zane-neo Is this targeted for 2.9?

**Comment by zane-neo** [comment]
> @zane-neo Can you also add this to the batch script version of [`install_demo_configuration.bat`](https://github.com/opensearch-project/security/blob/main/tools/install_demo_configuration.bat) as well?

done

**Comment by zane-neo** [comment]
> @zane-neo Is this targeted for 2.9?

Yes, this is targeted for 2.9.

**Review by cwperks — DISMISSED** [review]

**Review by willyborankin — DISMISSED** [review]

File: tools/install_demo_configuration.bat [hunk 1]
```diff
@@ -315,7 +315,7 @@ echo plugins.security.enable_snapshot_restore_privilege: true >> "%OPENSEARCH_CO
...
-echo plugins.security.system_indices.indices: [".plugins-ml-model", ".plugins-ml-task", ".opendistro-alerting-config", ".opendistro-alerting-alert*", ".opendistro-anomaly-results*", ".opendistro-anomaly-detector*", ".opendistro-anomaly-checkpoints", ".opendistro-anomaly-detection-state", ".opendistro-reports-*", ".opensearch-notifications-*", ".opensearch-notebooks", ".opensearch-observability", ".ql-datasources", ".opendistro-asynchronous-search-response*", ".replication-metadata-store", ".opensearch-knn-models"] >> "%OPENSEARCH_CONF_FILE%"
+echo plugins.security.system_indices.indices: [".plugins-ml-connector", ".plugins-ml-model", ".plugins-ml-task", ".opendistro-alerting-config", ".opendistro-alerting-alert*", ".opendistro-anomaly-results*", ".opendistro-anomaly-detector*", ".opendistro-anomaly-checkpoints", ".opendistro-anomaly-detection-state", ".opendistro-reports-*", ".opensearch-notifications-*", ".opensearch-notebooks", ".opensearch-observability", ".ql-datasources", ".opendistro-asynchronous-search-response*", ".replication-metadata-store", ".opensearch-knn-models"] >> "%OPENSEARCH_CONF_FILE%"
```
Comment (scrawfor99) [inline tools/install_demo_configuration.bat:318]: Is this supposed to be missing ".plugins-ml-model-group?" It is present in the Linux script but not the batch file.

Comment (zane-neo) [reply tools/install_demo_configuration.bat:318]: correct, this was missing, I'll add this as well.

**Review by scrawfor99 — APPROVED** [review]

**Review by cwperks — APPROVED** [review]

**Pull request merged**